tools/auth/webex.go

```diff
@@ -0,0 +1,80 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pocketbase/pocketbase/tools/types"
+	"golang.org/x/oauth2"
+)
+
+func init() {
+	Providers[NameWebex] = wrapFactory(NewWebexProvider)
+}
+
+var _ Provider = (*Webex)(nil)
+
+// NameWebex is the unique name of the Webex provider.
+const NameWebex string = "webex"
+
+// Webex allows authentication via Webex OAuth2.
+type Webex struct {
+	BaseProvider
+}
+
+// NewWebexProvider creates a new Webex provider instance with some defaults.
+func NewWebexProvider() *Webex {
+	return &Webex{BaseProvider{
+		ctx:         context.Background(),
+		displayName: "Webex",
+		pkce:        true,
+		scopes:      []string{"spark:people_read"},
+		authURL:     "https://webexapis.com/v1/authorize",
+		tokenURL:    "https://webexapis.com/v1/access_token",
+		userInfoURL: "https://webexapis.com/v1/people/me",
+	}}
+}
+
+// FetchAuthUser returns an AuthUser instance based on the Webex user api.
+//
+// API reference: https://developer.webex.com/docs/api/v1/people/get-my-own-details
+func (p *Webex) FetchAuthUser(token *oauth2.Token) (*AuthUser, error) {
+	data, err := p.FetchRawUserInfo(token)
+	if err != nil {
+		return nil, err
+	}
+
+	rawUser := map[string]any{}
+	if err := json.Unmarshal(data, &rawUser); err != nil {
+		return nil, err
+	}
+
+	extracted := struct {
+		Id          string   `json:"id"`
+		DisplayName string   `json:"displayName"`
+		UserName    string   `json:"userName"`
```

Comment:
There doesn't seem to be a `userName` field listed in the Webex `/v1/people/me` docs. Maybe it should be `nickName`?

Reply:
That seems to be an omission in the documentation then. I will let the product team know.
The `userName` field definitely exists for both internal auth (direct sign-ups) and external auth users in the response and is generally the user's email address.